fpdcc/ccfp-asset-dashboard
# Add Portfolio and PortfolioPhase models

## Overview

This PR adds models for persisting a Portfolio and its component Phases. I decided to break `PortfolioPhase` into a related model, rather than defining a `ManyToManyField` between Phase and Portfolio (I assume that a Phase could belong to any number of Portfolios, e.g., if it is not completed in a given year, or if Portfolios are used to compare different scenarios) because we needed somewhere to store a sequence for a Phase within a given Portfolio (which is distinct from its sequence in a `Project`).

This PR also introduces a `SequencedModel` abstract base model for sequencing phases within portfolios and projects. It should ensure that phases are created or updated at the specified sequence, and also update the sequence of downstream phases. I've added a test and docstrings for this functionality, but let me know if it's unclear.

Closes #85

## Testing Instructions

* Confirm that CI passes
* Confirm that a review app is created and released successfully
